engine/apps/public_api/serializers/incidents.py

```diff
@@ -42,10 +41,7 @@ class Meta:
         ]
 
     def get_alerts_count(self, obj):
-        return len(obj.alerts.all())
-
-    def get_title(self, obj):
-        return obj.alerts.all()[0].title
+        return obj.alerts_count
```

Comment:
nit: probably makes sense to add a comment that `alerts_count` is an annotated field which was added in `get_queryset`